cli/dataset.go

```diff
@@ -315,64 +317,58 @@ func binaryDataToJSONLines(ctx context.Context, client datapb.DataServiceClient,
 	}
 	datum := data[0]
 
-	// Make JSONLines
-	var jsonl interface{}
+	fmt.Println("Binary metadata: %v", datum.GetMetadata())
+	fmt.Println("Binary metadata file name: %v", datum.GetMetadata().GetFileName())
 
-	annotations := []Annotation{}
-	for _, tag := range datum.GetMetadata().GetCaptureMetadata().GetTags() {
-		annotations = append(annotations, Annotation{AnnotationLabel: tag})
-	}
-	classificationsAnnotations := datum.GetMetadata().GetAnnotations().GetClassifications()
-	for _, classification := range classificationsAnnotations {
-		annotations = append(annotations, Annotation{AnnotationLabel: classification.GetLabel()})
-	}
-	bboxAnnotations := convertBoundingBoxes(datum.GetMetadata().GetAnnotations().GetBboxes())
-
-	fileName := filepath.Join(dst, filenameForDownload(datum.GetMetadata()))
-	ext := datum.GetMetadata().GetFileExt()
-	// If the file is gzipped, unzip.
-	if ext != gzFileExt && filepath.Ext(fileName) != ext {
-		// If the file name did not already include the extension (e.g. for data capture files), add it.
-		// Don't do this for files that we're unzipping.
-		fileName += ext
+	jsonLines, err := mlTrainingClient.BinaryMetadataToJSONLines(ctx, &mltrainingpb.BinaryMetadataToJSONLinesRequest{
+		BinaryMetadata: []*datapb.BinaryMetadata{datum.GetMetadata()},
+		Path:           dst,
+	})
+	if err != nil {
+		return errors.Wrapf(err, "error converting binary metadata to JSON lines")
 	}
+	for _, jsonLine := range jsonLines.GetJsonLines() {
+		var data map[string]interface{}
+		if err := json.Unmarshal([]byte(jsonLine), &data); err != nil {
```

Comment:
Not a big fan of having to unmarshal and modify this JSON obj, will probably end up being pretty slow.